flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -242,6 +265,53 @@ public static StateBackend fromApplicationOrConfigOrDefault(
         return backend;
     }
 
+    public static StateBackend loadStateBackend(
+            @Nullable StateBackend fromApplication,
+            Configuration config,
+            ClassLoader classLoader,
+            @Nullable Logger logger)
+            throws IllegalConfigurationException, DynamicCodeLoadingException, IOException {
+
+        final StateBackend backend =
+                fromApplicationOrConfigOrDefault(fromApplication, config, classLoader, logger);
+
+        if (config.get(CheckpointingOptions.ENABLE_STATE_CHANGE_LOG)) {
+
+            Preconditions.checkArgument(
+                    backend instanceof DelegatedStateBackend, "backend is not delegable");
+
+            LOG.info(
+                    "Delegate State Backend is used, and the root State Backend is {}",
+                    backend.getClass().getSimpleName());
+
+            // ChangelogStateBackend resides in a separate module, load it using reflection
+            try {
+                Constructor<? extends DelegateStateBackend> constructor =
+                        Class.forName(CHANGELOG_STATE_BACKEND, false, classLoader)
+                                .asSubclass(DelegateStateBackend.class)
+                                .getConstructor(DelegatedStateBackend.class);
+
+                Class.forName(ROCKSDB_STATE_BACKEND_FACTORY, false, classLoader);
```

Comment:
Could you add a comment why is it needed?

Reply:
I've added it to test classloader, it should be removed. Sorry.

Reply:
Thanks for the careful review, it may cause bugs that are strange/difficult to debug.